packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -377,13 +405,34 @@ export function s(
         children = slotset[slotName];
     }
     const vnode = h('slot', data, children);
-    if (vnode.owner.renderer.syntheticShadow) {
+    if (vnode.owner.renderer.syntheticShadow || !hasShadow(vnode.owner)) {
         // TODO [#1276]: compiler should give us some sort of indicator when a vnodes collection is dynamic
         sc(children);
     }
+    if (!hasShadow(vnode.owner)) {
+        return fs(data, children);
+    }
     return vnode;
 }
 
+// [fs] fake slot
+export function fs(data: ElementCompilerData, children: VNodes): VFakeSlot {
+    const { key } = data;
+    let elm, text;
+    return {
+        sel: '',
+        data,
+        children,
+        text,
+        elm,
+        key,
+        hook: FakeSlotHook,
+        owner: getVMBeingRendered()!,
+        start: t(''),
+        end: t(''),
```

Comment:
This is really interesting! Using empty text nodes to mark the start/end of the slot boundary. Might make sense to add a comment here to explain this.

Reply:
I will like to understand better why are the markers needed. snabbdom doesn't really need the starting point, it only needs references to insert before. I was under the assumption that all the slot needs to do is to:

1. re-key the vnodes - which we do already for slot content due to default content vs real content
2. produce a vnodelist similar to iteration so it can be flatten by the regular process.

But after looking at this implementation, it seems that we are going into a different direction.